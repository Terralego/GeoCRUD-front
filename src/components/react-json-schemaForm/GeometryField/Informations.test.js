import React from 'react';
import renderer from 'react-test-renderer';
import { GeometryFieldContext } from './GeometryFieldProvider';

import Information from './Informations';

jest.mock('@terralego/core/modules/Map', () => ({}));

jest.mock('../../Message', () => props => <div {...props} />);

jest.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: text => text,
  }),
  Trans: () => text => text,
}));

describe('should snapshot correctly', () => {
  it('add required Point', () => {
    const tree = renderer
      .create(
        <GeometryFieldContext.Provider
          value={{
            isRequired: true,
            isRouting: false,
            nextFormData: {
              geom: {
                type: 'Point',
                coordinates: [],
              },
            },
          }}
        >
          <Information />
        </GeometryFieldContext.Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('edit required Point', () => {
    const tree = renderer
      .create(
        <GeometryFieldContext.Provider
          value={{
            isRequired: true,
            isRouting: false,
            nextFormData: {
              geom: {
                type: 'Point',
                coordinates: [1, 2],
              },
            },
          }}
        >
          <Information />
        </GeometryFieldContext.Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should manipulate routing', () => {
    const tree = renderer
      .create(
        <GeometryFieldContext.Provider
          value={{
            isRequired: false,
            isRouting: true,
            nextFormData: {
              geom: {
                type: 'LineString',
                coordinates: [1, 2],
              },
            },
          }}
        >
          <Information />
        </GeometryFieldContext.Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
