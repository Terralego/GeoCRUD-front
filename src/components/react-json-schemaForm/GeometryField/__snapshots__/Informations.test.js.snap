// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot correctly add required Point 1`] = `
Array [
  <p
    className="control-label"
  >
    <span>
      jsonSchema.geometryField.information-create
      <span
        className="required"
      >
        *
      </span>
    </span>
  </p>,
  <div
    className="geometry-field__message"
    intent="primary"
  />,
]
`;

exports[`should snapshot correctly edit required Point 1`] = `
Array [
  <p
    className="control-label"
  >
    <span>
      jsonSchema.geometryField.information-edit
      <span
        className="required"
      >
        *
      </span>
    </span>
  </p>,
  <div
    className="geometry-field__message"
    intent="primary"
  >
    jsonSchema.geometryField.helper-edit
  </div>,
]
`;

exports[`should snapshot correctly should manipulate routing 1`] = `
Array [
  <p
    className="control-label"
  >
    <span>
      jsonSchema.geometryField.information-edit
    </span>
  </p>,
  <div
    className="geometry-field__message"
    intent="primary"
  >
    jsonSchema.geometryField.helper-routing
  </div>,
]
`;
